libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendMessagingExtensionAuthResponse.ts

```diff
@@ -0,0 +1,187 @@
+} from 'adaptive-expressions';
+import {
+    ActionTypes,
+    Activity,
+    BotFrameworkAdapter,
+    CardAction,
+    ExtendedUserTokenProvider,
+    MessagingExtensionResult,
+    MessagingExtensionSuggestedAction,
+    TokenResponse,
+} from 'botbuilder';
+import {
+    Converter,
+    ConverterFactory,
+    Dialog,
+    DialogConfiguration,
+    DialogContext,
+    DialogTurnResult,
+} from 'botbuilder-dialogs';
+import { BaseSendTaskModuleContinueResponse } from './baseSendTaskModuleContinueResponse';
+
+export interface SendMessagingExtensionAuthResponseConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
+    property?: string | Expression | StringExpression;
+    connectionName?: string | Expression | StringExpression;
+    title?: string | Expression | StringExpression;
+}
+
+/**
+ * Send a messaging extension 'result' response when a Teams Invoke Activity is received with activity.name='composeExtension/queryLink'.
+ */
+export class SendMessagingExtensionAuthResponse
+    extends BaseSendTaskModuleContinueResponse
+    implements SendMessagingExtensionAuthResponseConfiguration {
+    /**
+     * Class identifier.
+     */
+    public static $kind = 'Teams.SendMessagingExtensionAuthResponse';
+
+    /**
+     * Gets or sets property path to put the value in.
+     */
+    public property: StringExpression;
+
+    /**
+     * Gets or sets the name of the OAuth connection.
+     */
+    public connectionName: StringExpression;
+
+    /**
+     * Gets or sets an Title of the response.
+     */
+    public title: StringExpression;
+
+    public getConverter(property: keyof SendMessagingExtensionAuthResponseConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'property':
+            case 'connectionName':
+            case 'title':
+                return new StringExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the dialog is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional, initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A promise representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: Record<string, unknown>): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled?.getValue(dc.state)) {
+            return dc.endDialog();
+        }
+
+        // Type check that dc.context.adapter matches interface, ExtendedUserTokenProvider
+        // eslint-disable-next-line no-prototype-builtins
+        if (!(typeof (dc.context.adapter as BotFrameworkAdapter)?.getUserToken === 'function')) {
```

Comment:
Use the `botbuilder-stdlib` assertions here. 

```typescript
import {assert} from 'botbuilder-stdlib';

...
const assertBotFrameworkAdapter: Assertion<BotFrameworkAdapter> = assert.instanceOf(BotFrameworkAdapter, 'BotFrameworkAdapter');
assertBotFrameworkAdapter(dc.context.adapter, ['dc', 'context', 'adapter']);
```

after the assertion, the compiler will actually understand that `dc.context.adapter` is a `BotFrameworkAdapter` so you won't have to do any extra type checking.